web/src/components/upload-indicator/upload-indicator.tsx

```diff
@@ -0,0 +1,33 @@
+import { FC, useContext, useEffect } from 'react';
+import { IndeterminateBar, ProgressBar } from '@epam/loveship';
+import { UploadIndicatorContext } from './upload-indicator.context';
+import styles from './upload-indicator.module.scss';
+
+export const UploadIndicator: FC<{ isRequestOngoing: boolean }> = ({ isRequestOngoing }) => {
+    const { progress, isUploaded, resetUploadProgressState } = useContext(UploadIndicatorContext);
+
+    const Indicator = isUploaded ? (
+        <IndeterminateBar />
+    ) : (
+        <ProgressBar progress={progress} size="12" />
+    );
+
+    const message = isUploaded ? 'Processing' : 'Uploading';
+
+    const indicatorWrapperStyle = isUploaded ? styles.processing : styles.uploading;
+
+    useEffect(() => {
+        if (progress === 100 && isUploaded && !isRequestOngoing) {
```

Comment:
isRequestOngoing corresponds from isLoading from outside component. In the outside component isLoading is set as false when the request finishes (see documents-page-control-connector@handleAddFiles)

isRequestOngoing does not tell if the file is uploading or not. As the name suggests it points if the request is ongoing. So it might be the case where progress === 100 (or isUploaded = true), i.e., file is uploaded, but still under processing, i.e., request is not finished

Reply:
Anyways, logic was removed from here because state is already being reseted at the custom fetch function

Reply:
Checked